mne/time_frequency/tfr.py

```diff
@@ -641,12 +794,42 @@ def _tfr_aux(method, inst, freqs, decim, return_itc, picks, average,
             # if the input is of class Evoked
             meta = evs = ev_id = None
 
-        out = EpochsTFR(info, power, times, freqs, method='%s-power' % method,
-                        events=evs, event_id=ev_id, metadata=meta)
-
+        if isinstance(inst, _BaseSourceEstimate):
+            out = _create_stfr(inst, power, freqs, method='%s-power' % method)
+        else:
+            out = EpochsTFR(info, power, times, freqs, events=evs,
+                            event_id=ev_id, metadata=meta,
+                            method='%s-power' % method)
```

Comment:
all this `if isinstance(inst, _Base..)` look alike. they either call `_create_stfr` or `FoobarTFR`. It seems that all this can get refactor.